ipa-core/src/helpers/transport/mod.rs

```diff
@@ -323,6 +341,31 @@ pub trait Transport: Clone + Send + Sync + 'static {
         route: R,
     ) -> Self::RecordsStream;
 
+    /// Broadcasts a message to all peers, excluding this instance, collecting all failures and
+    /// successes. This method waits for all responses and returns only when all peers responded.
+    /// The routes and data will be cloned.
+    #[allow(clippy::disallowed_methods)]
+    async fn broadcast<Q, S, R, D>(
+        &self,
+        route: R,
+        data: D,
+    ) -> Result<(), BroadcastError<Self::Identity, Self::Error>>
+    where
+        Option<QueryId>: From<Q>,
+        Option<Gate>: From<S>,
+        Q: QueryIdBinding,
+        S: StepBinding,
+        R: RouteParams<RouteId, Q, S> + Clone,
+        D: Stream<Item = Vec<u8>> + Clone + Send + 'static,
+    {
+        try_join_all(self.peers().map(|peer| {
```

Comment:
I love that it is a default implementation on this trait, `try_join_all` can be problematic though as it aggressively polls every future and is not optimized for large iterators. For MPC it is not a big deal, for shard connections it can be problematic. Given that you don't care about ordering here, `FuturesUnordered` is a better default imo

self.peers().map(|peer| {
            self.send(peer, route.clone(), data.clone())
                .map_err(move |source| BroadcastError { peer, source })
        })).collect::<FuturesUnordered>().await?;